src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionEndpointIT.java

```diff
@@ -0,0 +1,183 @@
+package uk.gov.ons.ctp.response.action.endpoint;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import java.io.IOException;
+import java.util.Random;
+import java.util.UUID;
+import javax.xml.bind.JAXBContext;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import uk.gov.ons.ctp.common.utility.Mapzer;
+import uk.gov.ons.ctp.response.action.config.AppConfig;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanPostRequestDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPostRequestDTO;
+import uk.gov.ons.ctp.response.casesvc.message.notification.CaseNotification;
+import uk.gov.ons.ctp.response.casesvc.message.notification.NotificationType;
+import uk.gov.ons.tools.rabbit.Rabbitmq;
+import uk.gov.ons.tools.rabbit.SimpleMessageListener;
+import uk.gov.ons.tools.rabbit.SimpleMessageSender;
+
+/** Integration tests for action endpoints */
+@Slf4j
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ActionEndpointIT {
+
+  @Autowired private ResourceLoader resourceLoader;
+
+  @LocalServerPort private int port;
+
+  @Autowired private ObjectMapper mapper;
+
+  @Autowired private AppConfig appConfig;
+
+  @Rule public WireMockRule wireMockRule = new WireMockRule(options().port(18002));
+
+  @ClassRule public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+  @Rule public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+  private Mapzer mapzer;
+
+  @Before
+  public void setup() {
+    mapzer = new Mapzer(resourceLoader);
+    initialiseUnirestObjectMapper();
+  }
+
+  @Test
+  public void ensureSampleIDExposed() throws Exception {
```

Comment:
I don't really follow what this is testing as there are no assertions and looks like we're making 3 write type operations with just a 201 assertion after the first request

Reply:
I think this one is fixed ... the assertions on the final message out check that the data contained is as expected.  The parts in the middle are waiting for stuff to complete or poking the action service to do more stuff so only asserting whether it's finished or works, but let me know if you still think it's light

Reply:
Some context might help in the test name maybe something like `ensureSampleIDIsExposedInActionRequest`

Reply:
Done